test/cpp/end2end/grpc_authz_end2end_test.cc

```diff
@@ -691,45 +680,39 @@ TEST_F(GrpcAuthzEnd2EndTest, StaticInitWithAuditLoggingOnDenyAndAllow) {
   grpc::Status status;
 
   ClientContext context1;
+  // Matches the allow rule.
   context1.AddMetadata("key-foo", "foo");
-  ::testing::internal::CaptureStdout();
   status = SendRpc(channel, &context1, &resp);
-  CompareAuditLogEntry(
-      ::testing::internal::GetCapturedStdout(),
-      "{\"authorized\":true,\"matched_rule\":\"allow_foo\",\"policy_name\":"
-      "\"authz\","
-      "\"principal\":\"spiffe://foo.com/bar/baz\",\"rpc_method\":\"/"
-      "grpc.testing.EchoTestService/Echo\"}");
   EXPECT_TRUE(status.ok());
-  EXPECT_EQ(log_count_, 1);
+  EXPECT_EQ(audit_logs_.size(), 1);
 
   ClientContext context2;
+  // Does not match the allow rule.
   context2.AddMetadata("key-foo", "bar");
-  ::testing::internal::CaptureStdout();
   status = SendRpc(channel, &context2, &resp);
-  CompareAuditLogEntry(
-      ::testing::internal::GetCapturedStdout(),
-      "{\"authorized\":false,\"matched_rule\":\"\",\"policy_name\":\"authz\","
-      "\"principal\":\"spiffe://foo.com/bar/baz\",\"rpc_method\":\"/"
-      "grpc.testing.EchoTestService/Echo\"}");
   EXPECT_EQ(status.error_code(), grpc::StatusCode::PERMISSION_DENIED);
   EXPECT_EQ(status.error_message(), "Unauthorized RPC request rejected.");
-  EXPECT_EQ(log_count_, 2);
+  EXPECT_EQ(audit_logs_.size(), 2);
 
   ClientContext context3;
-  context3.AddMetadata("key-foo", "foo");
+  // Matches the deny rule.
   context3.AddMetadata("key-bar", "bar");
-  ::testing::internal::CaptureStdout();
   status = SendRpc(channel, &context3, &resp);
-  CompareAuditLogEntry(
-      ::testing::internal::GetCapturedStdout(),
-      "{\"authorized\":false,\"matched_rule\":\"deny_bar\",\"policy_name\":"
-      "\"authz\","
-      "\"principal\":\"spiffe://foo.com/bar/baz\",\"rpc_method\":\"/"
-      "grpc.testing.EchoTestService/Echo\"}");
   EXPECT_EQ(status.error_code(), grpc::StatusCode::PERMISSION_DENIED);
   EXPECT_EQ(status.error_message(), "Unauthorized RPC request rejected.");
-  EXPECT_EQ(log_count_, 3);
+  EXPECT_EQ(audit_logs_.size(), 3);
+  EXPECT_THAT(
+      audit_logs_,
+      ::testing::ElementsAre(
+          "{\"authorized\":true,\"matched_rule\":\"allow_foo\",\"policy_"
```

Comment:
Same comment about checking the content after the associated RPCs, feels cleaner that way

Reply:
Done.